expo-package/src/index.js

```diff
@@ -1,7 +1,31 @@
+/**
+ * In Expo 34, all the modules such as image picker, document picker, permissions etc etc have been moved to their own packages.
+ * And they have been removed from Expo package. But these packages don't work with Expo 32 sdk (because of linking issue).
+ */
 import { registerNativeHandlers } from 'stream-chat-react-native-core';
 import { NetInfo } from 'react-native';
-// import { ImagePicker, Permissions } from 'expo';
-import { ImagePicker, DocumentPicker, Permissions } from 'expo';
+import { Constants } from 'react-native-unimodules';
+import * as Expo from 'expo';
+
+let ImagePicker;
+let DocumentPicker;
+let Permissions;
+let manifest = {};
+
+manifest = Constants.manifest;
+if (!manifest) {
+  manifest = Expo.Constants.manifest;
+}
+
+if (manifest.sdkVersion.split('.')[0] >= 33) {
+  ImagePicker = require('expo-image-picker');
+  Permissions = require('expo-permissions');
+  DocumentPicker = require('expo-document-picker');
+} else {
+  ImagePicker = Expo.ImagePicker;
```

Comment:
Another approach to check if ImagePicker exist in Expo package, would be to surround this statement with `try-catch`. But Error thrown by Expo package (when someone tries to access this package `ImagePicker` from `Expo`) is within timeout statement ... and thus doesn't really get caught on our end and it ends up displaying big red error message on screen.